### DISCUSSION TITLE
Preview Mod Look

### DISCUSSION THREAD

**Discussion by Arnasja** [xivdev/Penumbra#174] [Ideas] [answered]
When importing a mod there needs to be a preview option to let you know what it will like.
I wanted to post this in the Suggestion area but could not locate it.
An example is showed in the attached photo.
![Preview](https://user-images.githubusercontent.com/102417617/160614005-ab04682c-ee7a-4a9f-a1f1-6f8ef382eb2e.png)

**Comment by Ottermandias** [comment]
No, there does not need to be such an option. You are ingame. You can check it out yourself.

**Reply by Arnasja** [reply]
Before I commit to installing a mod, I want to see what it looks like. That's the main reason I won't switch to Penumbra. Unless I am wrong and there is a way to preview in game.

**Reply by reiichi001** [reply]
> Before I commit to installing a mod, I want to see what it looks like. That's the main reason I won't switch to Penumbra. Unless I am wrong and there is a way to preview in game.

TexTools doesn't let you preview a mod either, and requires you to import it and launch the game too.

Or are you asking for a display for any included images that might have been included with the modpack? Because that's what the textools "preview" is. It's not rendering the mod. It's a static image. Given that Penumbra imports a mod immediately because you don't go through an install wizard, that would be a bit of a different workflow for imports. One-time choices in textools aren't the case in penumbra because you can dynamically enable/disable them on the fly.

**Reply by Ottermandias** [reply]
So, you download the mod from whatever, ignore what it does, but do not want to 'commit' to installing it? Because... just deleting it again if you did not like it is three clicks too many?

**Reply by reiichi001** [reply]
For me personally, no, because I already know what I want before I download it, and most of the mods I want have their preview images on XIVModArchive or wherever I'm getting them. 

The modpack's preview images could possibly be useful before unpacking a large multi-GB sized modpack, but I don't think it's worth implementing just because that's how textools does it. I much prefer Penumbra installing the mod and making it accessible as soon as possible, especially since as described above, it doesn't have the same import mod wizard requirements (really, limitations) as Textools.

**Reply by Arnasja** [reply]
Yes, it's the preview.  This is what I wanted to know. You two have brought valid arguments in favor of Penumbra. You've convinced me to give another go. Thank you.
One other question though. Is there a guide somewhere that explains the various options?

**Reply by Arnasja** [reply]
Okay. I gave Penumbra another try but every time I try to use a clothing mod I get weird skin textures. I really like the idea of being able to toggle mods off and on but if I try to use a mod and it doesn't work correctly then what's the use.
Oh BTW... I only had Bibo+ installed when I tried the clothing mod. Maybe next release will be better and I probably think I should start posting my problems in another sextion :)
Picture attached on skin textures.
![Body Texture](https://user-images.githubusercontent.com/102417617/160835022-a20b0ccb-332f-47ba-8f26-0ceb44623643.png)

**Comment by Ottermandias** [comment, marked as answer]
That has nothing to do with Penumbra not working correctly. In general, if your skin is changed at all, it's not on Penumbra if things look wonky. Penumbra only tells the game to load other files, not how. 
You're using a mod that is not inherently setup for bibo skin materials with bibo skin materials. You can enable advanced mode in the config, then edit mode on the clothing mod, and then change From Material Suffix... (enter "b" here, I assume) To Material Suffix... (enter "bibo" here, I assume) and check if that works.

**Reply by Arnasja** [reply]
I'm inclined to believe that Penumbra is more for someone that knows what they are doing and I know just enough to be dangerous.
I'll give your suggestions a try. I need to experiment a bit more before throwing my hands in the air. I really want this to work. Believe me.
Thanks for all you help.

**Reply by reiichi001** [reply]
Textools has created what one could only reasonably call "a bunch of really fucked up shit" between poor design decisions, poor handling of those decisions, and some absolutely terrible lack of validation for mod packagers, which then makes everything worse. 

Penumbra aims to not put up with that because it leads to bad practices and very terrible issues to debug, like the material conversion issue noted above. (This wouldn't be a problem if the mod maker wasn't allowed to specify arbitrary material files that don't otherwise exist in the game data and then hardcoding their mods to use them while other base body/skin competing mods use their own material files wth another different name.)

In textools, it might just explode. Or worse, make incorrect assumptions, or even worse, just overwrite any data it doesn't like anyways because it doesn't care about anything being done correctly if it can just smash through it. And then people wonder how and why they get a shiba inu minion picture instead of their minimap when their index files have been so mangled that it's a wonder the game even runs.

**Reply by Arnasja** [reply]
Just to let you know @Ottermandias That your suggestion to convert the Material worked perfectly. No weird skin issues so far :)
and @reiichi001 Looks like I'll be sticking with Penumbra. it makes it really nice to be able to enable an unable a mod on the fly. Don't want to be streaming and have some TOS clothing mod come up on my stream ;)
Thank you both for all your help. <3

**Reply by Arnasja** [reply]
Just want to say one more thing before I put this to rest... I hate you guys ( **not really** ) for making me a believer in  Penumbra.
It's so convenient to have mods turned off and turn them on when needed and also if a mod has multiple options, just select the ones that I want without being committed to the ones installed through TextTools. With that other program, that won't be mentioned again, I had exit FFXIV rerun the install mod option and select different options. Now I don't have to :)
I ramble. Hope what I wrote makes sense.
I can't thank you enough for all your help.
Until I pester you again.
<3 Arna
